src/Scene/TreeUpdater.js

```diff
@@ -0,0 +1,151 @@
+    var positionWorld = new THREE.Vector3();
+
+    return function (node, camera, sceneParams) {
+        node.setMatrixRTC(camera.getRTCMatrixFromCenter(positionWorld.setFromMatrixPosition(node.matrixWorld)));
+        node.setFog(sceneParams.fogDistance);
+        node.setSelected(sceneParams.selectedNodeId === node.id);
+    };
+}());
+
+TreeUpdater.prototype._selectNode = function (node, id) {
+    if (node.id === id) {
+        /* eslint-disable no-alert, no-console */
+        console.info(node);
+        /* eslint-enable no-alert, no-console */
+    } else {
+        applyFunctionToChildren(n => this._selectNode(n, id), node);
+    }
+};
+
+function applyFunctionToChildren(func, node) {
+    for (let i = 0; i < node.children.length; i++) {
+        func(node.children[i]);
+    }
+}
+
+/**
+ * @documentation: Initiate traverse tree
+ * @param {type} tree       : tree
+ * @param {type} camera     : current camera
+ * @param {type} process    : the process to apply to each node
+ * @param {type} optional   : optional process
+ * @returns {undefined}
+ */
+TreeUpdater.prototype.browse = function (tree, camera, process, layersConfig, sceneParams) {
+    camera.updateMatrixWorld();
+
+    var fogDistance = sceneParams.fogDistance * Math.pow((camera.getDistanceFromOrigin() - 6300000) / 25000000, 1.6);
+    sceneParams.fogDistance = fogDistance;
+
+    process.prepare(camera);
+
+    var params = {
+        sceneParams,
+        tree,
+        layersConfig,
+        fogDistance,
+    };
+
+    var rootNode = tree.children[0];
+    applyFunctionToChildren(n => this._browseDisplayableNode(n, camera, process, params), rootNode);
+};
+
+/**
+ * @documentation: Recursive traverse tree
+ * @param {type} node       : current node
+ * @param {type} camera     : current camera
+ * @param {type} process    : the process to apply to each node
+ * @param {type} optional   : optional process
+ * @returns {undefined}
+ */
+TreeUpdater.prototype._browseDisplayableNode = function (node, camera, process, params) {
+    if (node.parent.isVisible() && process.processNode(node, camera, params)) {
+        if (node.isDisplayed()) {
+            this.uniformsProcess(node, camera, params.sceneParams);
+            applyFunctionToChildren(n => this._browseNonDisplayableNode(n, node.level + 2, process, camera, params), node);
+        } else {
+            applyFunctionToChildren(n => this._browseDisplayableNode(n, camera, process, params), node);
+        }
+    } else {
+        node.setVisibility(false);
+        node.setDisplayed(false);
+
+        applyFunctionToChildren(n => this._browseNonDisplayableNode(n, node.level + 2, process, camera, params), node);
+    }
+};
+
+TreeUpdater.prototype._browseNonDisplayableNode = function (node, level, process, camera, params) {
+    // update node's sse value
+    node.sse = camera.computeNodeSSE(node);
+    node.setDisplayed(false);
+
+    var sse = process.checkNodeSSE(node);
```

Comment:
This `TreeUpdater` could be modified to work with any `Layer`.
It just needs to recursively traverse the tree until `process()` function returns false.
And all the visibility, SSE stuff should move to helper classes or Globe processor.